[ci skip] Clarify maintenance policy for bug fixes to security patches (#41524)
rails/rails: guides/source/maintenance_policy.md

security releases are easy to upgrade to if you're running the latest version
of Rails.

> Only direct security patches will be included in security releases. Fixes to
> address non-security related bugs related to a security patch may be published

Reviewer: Small tweak for easier parsing:

```suggestion
Only direct security patches will be included in security releases. Fixes for
non-security related bugs resulting from a security patch may be published
```

> non-security-related bugs or other implications of a security patch may be published

Reviewer 2: ```suggestion
non-security related bugs related to a security patch may be published
```

Author: Thanks for the speedy review!

Just to clarify the thought behind "other implications", I would generally understand "bug" to mean "unintended", whereas my impression (which may be wrong) is that the changes in #38063 to make the `TestSession` behaviour consistent were _knowingly_ left out of the original security patch due not being directly necessary to fix the security issue at hand.

Reviewer 2: I assumed this was related to https://github.com/rails/rails/commit/892eab777c418135ce0646e91bc9ebb08a29ab9b not being included in the security release for https://github.com/rails/rails/commit/2a52a38cb51b65d71cf91fc960777213cf96f962. This, to me, is an unintended bug introduced by the security patch because the security patch was made without considering Ruby 2.2 requirements.

https://github.com/rails/rails/pull/38063 appears to be a side effect of the security patch too, which is knowingly left out for the same reason the syntax error fix was. I think it is implied that subsequent changes wouldn't be included in another release of 5.2.x. Without the full context, I think the document is more concise not mentioning other implications.

Author: On reflection, I think the point is still made sufficiently without the "other implications". I'll amend the commit with the suggested changes now.

> release's x-y-stable branch, and will be only be released as a new gem in

Reviewer 2: ```suggestion
release's x-y-stable branch, and will only be released as a new gem in
```
One last thing, then this LGTM.

Author: Great spot! Fixed

> on a series' X-Y-stable branch on git, but will be only be released as a new gem in

Reviewer 2: ```suggestion
on a release's x-y-stable branch, and will be only be released as a new gem in
```
To match the style of the paragraph above.

Author: I agree with most of this suggestion, but I thought the capitalised `X-Y-stable` branch name better matched the style of the rest of the document?
